### PR TITLE
ci: add codespell

### PR DESCRIPTION
**Pull request by Rakshith-R**
Adds codespell github actions

Updates: #12

### CODE REVIEW

**Review by yati1998 — APPROVED** [review]
LGTM

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+
```
Comment (nixpanic) [inline .github/workflows/golangci-lint.yml:1]: Unneeded empty line. Better start with `---` on the first line as this is a yaml file.

File: .github/workflows/golangci-lint.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      with:
+        # Optional: version of golangci-lint to use in form of v1.2 or v1.2.3 or `latest` to use the latest version
+        version: latest
+
+        # Optional: working directory, useful for monorepos
+        # working-directory: somedir
+
+        # Optional: golangci-lint command line arguments.
+        # TODO: remove --skip-files controllers/suite_test.go once logic is implemented
+        args: -E gosec --timeout=6m --skip-files controllers/suite_test.go
```
Comment (nixpanic) [inline .github/workflows/golangci-lint.yml:26]: why `-E gosec`?

**Review by nixpanic — CHANGES_REQUESTED** [review]

**Comment by Rakshith-R** [comment]
Added only codespell 
golangci linter need some extensive changes and config files(to ignore a few linters)

**Review by nixpanic — APPROVED** [review]

**Review by yati1998 — APPROVED** [review]

**Comment by Rakshith-R** [comment]
manually merging, since this modifies mergify.yml

**Pull request merged**